Automattic/wp-calypso
# Seller Experience: Add upgrade modal behind new feature flag

#### Proposed Changes

* When `signup/seller-upgrade-modal` feature flag is enabled, show an upgrade modal when a user clicks on "Unlock theme" when previewing a premium theme

<img width="808" alt="Screen Shot 2022-06-22 at 5 41 53 PM" src="https://user-images.githubusercontent.com/2124984/175143842-25ab9aaa-3dd8-43ed-9469-95b68b5b52f0.png">


#### Testing Instructions

* Switch to this PR, navigate to `/setup/designSetup?siteSlug=yoursite&flags=signup/theme-preview-screen,signup/seller-upgrade-modal`
* Click on any premium theme
* Click "Unlock theme" to show the modal
* Cancel, the close button, or clicking outside the modal should close it
* Clicking Upgrade should take you to checkout with the Pro plan in the cart
* Tracks events with the theme as a prop should be recorded for each click (opening the modal, upgrading, and closing)
* Disabling the feature flag query param in the URL should take you directly to checkout instead of opening the modal

Related to #64783
